source/contracts/reporting/Market.sol

```diff
@@ -171,20 +169,44 @@ contract Market is DelegationTarget, Typed, Initializable, Ownable, IMarket {
     }
 
     function updateTentativeWinningPayoutDistributionHash(bytes32 _payoutDistributionHash) public returns (bool) {
+        if (_payoutDistributionHash == tentativeWinningPayoutDistributionHash) {
+            return true;
+        }
+
         IReportingToken _reportingToken = reportingTokens[_payoutDistributionHash];
         require(address(_reportingToken) != NULL_ADDRESS);
 
         IReportingToken _tentativeWinningReportingToken = reportingTokens[tentativeWinningPayoutDistributionHash];
         if (address(_tentativeWinningReportingToken) == NULL_ADDRESS) {
             tentativeWinningPayoutDistributionHash = _payoutDistributionHash;
-            _tentativeWinningReportingToken = _reportingToken;
+            return true;
         }
-        if (_reportingToken.totalSupply() > _tentativeWinningReportingToken.totalSupply()) {
+
+        int256 _tentativeWinningStake = getPayoutDistributionHashStake(tentativeWinningPayoutDistributionHash, _tentativeWinningReportingToken);
+        int256 _payoutStake = getPayoutDistributionHashStake(_payoutDistributionHash, _reportingToken);
+
+        if (_payoutStake > _tentativeWinningStake) {
+            secondPlaceTentativeWinningPayoutDistributionHash = tentativeWinningPayoutDistributionHash;
             tentativeWinningPayoutDistributionHash = _payoutDistributionHash;
         }
         return true;
     }
 
+    function getPayoutDistributionHashStake(bytes32 _payoutDistributionHash, IReportingToken _reportingToken) private returns (int256) {
+        int256 _payoutStake = int256(_reportingToken.totalSupply());
+        if (address(automatedReporterDisputeBondToken) != NULL_ADDRESS) {
+            if (automatedReporterDisputeBondToken.getDisputedPayoutDistributionHash() == _payoutDistributionHash) {
+                _payoutStake -= int256(Reporting.automatedReporterDisputeBondAmount());
+            }
+        }
+        if (address(limitedReportersDisputeBondToken) != NULL_ADDRESS) {
+            if (limitedReportersDisputeBondToken.getDisputedPayoutDistributionHash() == _payoutDistributionHash) {
+                _payoutStake -= int256(Reporting.limitedReportersDisputeBondAmount());
+            }
+        }
```

Comment:
For completeness, it feels like we should include `allReportersDisputeBond` even though we don't use this function directly during a fork at the moment.  The UI however may want to present the various stakes during a fork and they would use this function and the all-dispute should be included in that case.